### DISCUSSION TITLE
How do i set up SaveSession to save automatically to the currently open session?

### DISCUSSION THREAD

**Discussion by bokonist** [rmagatti/auto-session#137] [Q&A] [answered]
The problem i'm having is, I open a project folder, i navigate to some subfolder, or a plugin sets the cwd to some other dir when i use it (neorg in my case). After that, when I run SaveSession, I'd like it to just overwrite whatever session I opened last. 

Current behaviour is, SaveSession just writes the session file with the same name as cwd, which just clutters my session folder with a bunch of sessions i never wanted.

**Comment by rmagatti** [comment, marked as answer]
Something I've been meaning to implement for a while now is cwd tracking. It's now possible because Neovim provides a `DirChanged` and `DirChangedPre` aucmds.
https://sourcegraph.com/github.com/neovim/neovim/-/blob/src/nvim/auevents.lua

What this would do is track that the cwd changed within the plugin and then do something with that information. That something would probably be to ignore a SaveSession happening when a cwd changes to start with.

**Reply by bokonist** [reply]
thanks @rmagatti , i'll figure something out with this.

**Reply by rmagatti** [reply]
@bokonist this feature is implemented now btw!

**Reply by bokonist** [reply]
thank you so much @rmagatti

**Reply by howabouthis** [reply]
Do I understand correctly, that when you :cd to another directory, and that directory has an existing session, the windows should change into those of the new session? 

I've been trying to figure out how to do that.

**Reply by rmagatti** [reply]
@howabouthis yes, but not by default, you have to enable that feature with the `cwd_change_handling` option (the readme documents how to use it)

**Reply by howabouthis** [reply]
Is the default behavior without that feature that the current session will overwrite the stored session in the upcoming directory?

**Reply by rmagatti** [reply]
Yeah as it doesn't know that anything has changed session wise or current working directory wise so it assumes it's just the normal previously opened session.
